public/app/plugins/panel/heatmap/types.ts

```diff
@@ -0,0 +1,31 @@
+export {
+  HeatmapColorMode,
+  HeatmapColorScale,
+  HeatmapColorOptions,
+  YAxisConfig,
+  CellValues,
+  FilterValueRange,
+  HeatmapTooltip,
+  HeatmapLegend,
+  ExemplarConfig,
+  RowsHeatmapOptions,
+  PanelOptions,
+  PanelFieldConfig,
+} from './panelcfg.gen';
+
+import { defaultPanelOptions as defaultPanelOptionsGen, PanelOptions } from './panelcfg.gen';
+
+//avoid type assertion
+export const defaultPanelOptions: PanelOptions = {
+  ...defaultPanelOptionsGen,
+  calculation: defaultPanelOptionsGen.calculation!,
+  cellValues: defaultPanelOptionsGen.cellValues!,
+  color: defaultPanelOptionsGen.color!,
+  tooltip: defaultPanelOptionsGen.tooltip!,
+  legend: defaultPanelOptionsGen.legend!,
+  yAxis: defaultPanelOptionsGen.yAxis!,
+  exemplars: defaultPanelOptionsGen.exemplars!,
+  filterValues: defaultPanelOptionsGen.filterValues!,
+  rowsFrame: defaultPanelOptionsGen.rowsFrame!,
+  showValue: defaultPanelOptionsGen.showValue!,
+};
```

Comment:
```suggestion
// Assert all types to be required
export const defaultPanelOptions = defaultPanelOptionsGen as PanelOptions;
```
I looked a bit more into this and it seems that the issue here is that defaults are generated as partial types, while in this specific case all the properties are defined. I think using type assertion is a better workaround here that redeclaring the defaults.

Reply:
Yes, the issue is because of the partials and could be easily solved with a cast. I tried to avoid it so betterer doesn't get upset with me :D

But since I could overwrite the `defaultPanelOptions` enum values in `types.ts` wouldn't it be easier to just leave it all in there for now?

Reply:
Ignore the second half of the above. I'll modify and push soon :D Thank you!

Reply:
Okay, so I've imported the default values, modified the problematic enum fields and exported as `PanelOptions`. Tell me if it's okay or if I misunderstood anything.